### ISSUE TITLE
Improving theme styles

### ISSUE THREAD

**Issue by K4sum1** [Eclipse-Community/r3dfox#36]
This are some ideas for improving the browser theme.

1: Have TabsToolbar item color match Active/Inactive Title Bar font color in classic theme.
    This would make it readable or at the very least match in all scenarios with classic theme, instead of the current always white it is.

2: Improve 8 classic theme and Aero Glass compatibility.
    Classic theme has no buttons and looks off in other ways. Aero Glass seems mostly fine but has no fog therefore text can be hard to read. Not sure how to do this since current code should pick up on it but doesn't. Maybe it's restricted to 7 only and I just need to add 8 for it to work idk.

3: Replace fog with title bar/caption text glow?
    This would be a more faithful Aero like design, but I'm not sure if I would like how it looks. I would need to realize this before I know though.

**Comment by i486** [comment]
I'm testing r3dfox for the first time and I'm unsure if it has already been reported.

Using: [Release v125.0rc2](https://github.com/Eclipse-Community/r3dfox/releases/tag/125.0rc2)

The bookmark folder dropdown appears as follows:
![image](https://github.com/Eclipse-Community/r3dfox/assets/625325/1376d034-4793-4ecd-a7c3-74c5d22ef846)

Theme:
[Beautiful New Opera Dark](https://addons.mozilla.org/es/firefox/addon/beautiful-new-opera-dark/) from Mozilla Firefox's add-on store.

**Comment by i486** [comment]
How it looks in Firefox 115 ESR

![image](https://github.com/Eclipse-Community/r3dfox/assets/625325/9731b72b-59aa-4ee2-b075-2fd26561da53)

**Comment by i486** [comment]
Tab bar has translucency. Any quick fix to disable this behavior?

![image](https://github.com/Eclipse-Community/r3dfox/assets/625325/2373ee59-ea9b-419f-9200-f7f9cc22b06d)

Firefox 115 ESR

![image](https://github.com/Eclipse-Community/r3dfox/assets/625325/0dee4c71-9eef-4c13-9f50-16d677c6775a)

**Comment by K4sum1** [comment]
> The bookmark folder dropdown appears as follows:

It's intentional.

https://github.com/Eclipse-Community/r3dfox/commit/9597d9dff06960bf18fe5004854e3d4d35f38a20

The ideal way would be to do it only if non native controls are enabled, but I haven't figured out how. Also need to get rid of the border.

> Tab bar has translucency. Any quick fix to disable this behavior?

I have an idea, I'll try it for next release.

**Comment by i486** [comment]
> It's intentional.
> 
> [9597d9d](https://github.com/Eclipse-Community/r3dfox/commit/9597d9dff06960bf18fe5004854e3d4d35f38a20)
> 
> The ideal way would be to do it only if non native controls are enabled, but I haven't figured out how. Also need to get rid of the border.

Upon reviewing my initial post, I realize there might have been some lack of clarity. I was just referring to the color of the bookmark dropdown menu. Currently, it's appearing white on the dark theme, whereas I expected it to match the dark theme's color scheme. Could you confirm if this was intentional?

r3dfox (same dark theme)
![322687756-1376d034-4793-4ecd-a7c3-74c5d22ef846](https://github.com/Eclipse-Community/r3dfox/assets/625325/ec31fd1b-a036-47d3-81db-fa13178d40d0)

Firefox 115 ESR (same dark theme)
![322689894-9731b72b-59aa-4ee2-b075-2fd26561da53](https://github.com/Eclipse-Community/r3dfox/assets/625325/ff414f29-31db-4643-b35d-a5fcaca93c17)

**Comment by K4sum1** [comment]
Yes, the idea is to make it take after the system theme more. I should make it take after disabling the non-native theme, but I'm not sure how.

**Comment by K4sum1** [comment]
So I redid the implementation, it will only be like that with native menus. However I don't think there's a way to disable native menus on 7/8. Maybe someone could pr it.

**Comment by garry-ut99** [comment]
> > The bookmark folder dropdown appears as follows:
> 
> It's intentional.
> 
> [9597d9d](https://github.com/Eclipse-Community/r3dfox/commit/9597d9dff06960bf18fe5004854e3d4d35f38a20)
> 
> The ideal way would be to do it only if non native controls are enabled, but I haven't figured out how. Also need to get rid of the border.

I prefer the previous look of bookmark menu, because the classic looks like shitty grey toilet paper, how to restore the previous look, is it possible via `userChrome.css`, I've tried to copy&paste the removed lines (marked as red) from the commit to `userChrome.css` and restarted browser but no effect, despite I have `toolkit.legacyUserProfileCustomizations.stylesheets` already set to `true`.

**Comment by K4sum1** [comment]
You might need to add some !important stuffs, but it should work.

**Comment by garry-ut99** [comment]
Ok, `!important` has worked.

**Comment by i486** [comment]
I was about to report this

> There seems to be a bug in the private window tab bar.
> It looks like Redfox's red background from private browsing is overlaid on the current theme's background in the tab bar.
> 
> Normal window
> ![image](https://github.com/Eclipse-Community/r3dfox/assets/625325/c2a93642-76c5-4bd3-96c5-9fa4adadd7cf)
> 
> Private window
> ![image](https://github.com/Eclipse-Community/r3dfox/assets/625325/110d1820-8d0d-4cd0-bf6d-79ccbdd5cb4c)

Then I read the following, which helped me realize it's intentional, prompting me to read more and write the comment below (ChatGPT'd due to my English; sorry for any cringe, but it helps convey my thoughts effectively).

> 4: Red tint on titlebar in private mode?
> I had this in a very old userChrome.css I made for Firefox 68 and the idea seems cool. Would make it even easier to tell private mode apart from normal mode. Done

As someone who is not a developer and also not well-versed in English. I use translation tools and AI assistants to understand and write online posts. In my initial exploration of R3dfox, due to my limited programming knowledge, much of the UI and developer jargon remained beyond my understanding, so I often skim over them as you can see above when I nearly reported a "bug" on private window tint.



Anyways, before downloading R3dfox, I read several of your posts mentioning your intention to preserve a largely stock Firefox experience, with Windows 7 support. This led me to assume (perhaps prematurely) that most of the discussions in the Issues section and the some discrepancies I've encountered in R3dfox were primarily due to ongoing efforts to reinstate or fix issues removed or broken by Firefox onward 115+ due to them removing Windows 7 support. While this undoubtedly applies in many instances in r3dfox, further reading in issues section helped me realize that significant portion of the posts to be related to your own enhancements and modifications, which I completely respect given the nature of a personal fork.

But still, it would be highly appreciated if there were options available for those seeking a stock experience. As what some may perceive as an enhancement, others may view as a drawback.

For example, garry-ut99 describes the bookmark folder menu as "shitty grey toilet paper," I find it a nice improvement over the appearance of Firefox 115's bookmark menu, especially now with the removal of odd borders.

Similarly, some might appreciate the red tint on the tab bar as a clear indicator of private browsing. I initially mistook it for a hardware issue. In certain theme, it looks just like the initial symptom of a how my GPU died.

Within the issue reports, someone suggested using VxKex for those that want stock Firefox. While an option, VxKex is unstable with the latest Firefox; it crashes after a brief period for me. This effectively renders R3dfox the sole viable choice for Windows 7 users who want to use new Firefox. Also there will be many people like me who like certain aspects of r3dfox and would prefer to use r3dfox over original Firefox.

Therefore, I propose the implementation of a configuration system utilizing the about:config (or something easier) functionality. This would allow users to restore r3dfox specific customization through a series of configurable options. Examples:

    r3dfox.customization.bookmarkfoldermenu: true (classic, default) / false (modern, original)
    r3dfox.customization.titlebar.tintonprivatewindow: true (default) / false (original)

While the development effort required for such a system is uncertain for me, I firmly believe it would be a valuable addition, potentially reducing the volume of future reports within the issues section.

What do you think?

**Comment by K4sum1** [comment]
I figured out how to make it get controlled by the preference I wanted it to be. Enabling the non-native theme brings the stock styles back now.

![ScreenShot_20240420095336](https://github.com/Eclipse-Community/r3dfox/assets/77039240/15bed6bd-8ce8-4043-b2da-7c647176fc88)

Going to implement private bar stuff now.

**Comment by K4sum1** [comment]
Done. Will try to add GUI option now.

![ScreenShot_20240420102803](https://github.com/Eclipse-Community/r3dfox/assets/77039240/06b69f94-e501-498c-a15a-14fc0c91e7f7)

**Comment by K4sum1** [comment]
> Within the issue reports, someone suggested using VxKex for those that want stock Firefox. While an option, VxKex is unstable with the latest Firefox; it crashes after a brief period for me.

I thought about this. What if you add this to Firefox prefs.js and try again?
`user_pref("browser.taskbar.lists.legacyBackend", true);`

**Comment by garry-ut99** [comment]
> Within the issue reports, someone suggested using VxKex for those that want stock Firefox. While an option, VxKex is unstable with the latest Firefox; it crashes after a brief period for me.

Confirmed , looking at VxKex issue tracker, there are many problem reports:
- https://github.com/vxiiduu/VxKex/issues/198
- https://github.com/vxiiduu/VxKex/issues/184
- https://github.com/vxiiduu/VxKex/issues/155
- https://github.com/vxiiduu/VxKex/issues/140
- https://github.com/vxiiduu/VxKex/issues/103
- https://github.com/vxiiduu/VxKex/issues/85

**Comment by K4sum1** [comment]
Should mention, the options have come in 125.0.2-2.

**Comment by i486** [comment]
> Coming in next release
> 
> ![image](https://private-user-images.githubusercontent.com/77039240/324192892-30c7cc56-f3d6-4e94-9530-f3775bcdf4d9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NDI2NzAsIm5iZiI6MTcxMzc0MjM3MCwicGF0aCI6Ii83NzAzOTI0MC8zMjQxOTI4OTItMzBjN2NjNTYtZjNkNi00ZTk0LTk1MzAtZjM3NzViY2RmNGQ5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDIxVDIzMzI1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdjMDA3ZmJhMjlmOTFkZjgwNjgwZThlNjJjMTcwYTE1MTFiMTkyYWM0YmYwZGYyY2FhODhjYjJiODIyMzIwMGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.3zWPjLcy8Nk-8fmCbyEiNzxzQ6k6EPp6vWbNtqLgEq0)

Wow, I wasn't anticipating real options. I was simply hoping for new about:config configurations, allowing users who prefer stock Firefox to insert them themselves. This is great!

**Comment by i486** [comment]
> So I didn't fix the tab bar translucency, but I think I know how.
> 
> I make my latest commit only apply to non-lwtheme, and then force it transparent in light dark overrides.

Tab bar translucency appears to be fixed. Really appreciated the great work!

**Comment by K4sum1** [comment]
Yeah, I figured I would add it as an option because I wanted to do that eventually anyways. It wasn't that bad.

**Comment by i486** [comment]
> > Within the issue reports, someone suggested using VxKex for those that want stock Firefox. While an option, VxKex is unstable with the latest Firefox; it crashes after a brief period for me.
> 
> I thought about this. What if you add this to Firefox prefs.js and try again? `user_pref("browser.taskbar.lists.legacyBackend", true);`

I believe this configuration resolved my issue of Firefox 125 crashing after a while. r3dfox is already my primary browser, so I won't be using Firefox since I encountered some brief freezing during a quick test and also find the caption looks ugly. Nevertheless, I'll keep it for testing consistency between same version of r3dfox and Firefox. Thanks!

**Comment by i486** [comment]
Bug: When we initiate the "Private Window," a split-second flash of the original Firefox's purple background is visible. This flash should be disabled to ensure the r3dfox's private red window opens seamlessly.

Feature idea: Regarding private windows, unchecking "Use r3dfox theme styles" should also include restoring the stock purple background in the Private window.

**Comment by K4sum1** [comment]
I fixed both, and it will come next release. (probably 125.0.3)

Waiting for official 125.0.3 to release.

**Comment by i486** [comment]
When using a non-native stock style, we have this messy separator line at the end of the dropdown menu of bookmark folders.

`r3df0x 125.0.3 & 126.0-2`
![image](https://github.com/Eclipse-Community/r3dfox/assets/625325/5b0418d6-8a59-4ceb-a181-2b975d40b3a1)

Here is how it's supposed to look:
`Firefox 126.0 (VxKex)`
![image](https://github.com/Eclipse-Community/r3dfox/assets/625325/a9dd9218-7d8b-436b-bf04-a7afc3ab1b3e)

**Comment by K4sum1** [comment]
That's related to menu separators. That code is hell. It could probably be fixed, but I'm not doing it.

**Comment by i486** [comment]
Do you know if it's possible to fix this via userChrome.css? If so, I plan to ask around in the userChrome.css community (subreddit, forum, etc.). This would require some convincing on my part since they'd need to install r3dfox to investigate.

@garry-ut99
Do you still see the same buggy separator line in your bookmark dropdown menu? I'm interested because you're using userChrome.css to restore the stock look.
https://github.com/Eclipse-Community/r3dfox/issues/36#issuecomment-2065387213

**Comment by K4sum1** [comment]
It is all CSS, so probably. I assume just override the changes here with !important.
https://github.com/Eclipse-Community/r3dfox/commit/894dd2406158342b699ef9dd146877ab9bd0f370

**Comment by garry-ut99** [comment]
> Do you still see the same buggy separator line in your bookmark dropdown menu?

Yes, but only in dark themes.

**Comment by i486** [comment]
I wasn't able to fix it by writing rules from the commit, as I lack CSS knowledge. However, things started to make a bit more sense after I discovered the Browser Toolbox and turned to ChatGPT for assistance. 

Here's a fix for anyone interested in giving it a try.

```
@media (-moz-bool-pref: "widget.non-native-theme.enabled") {
  #PlacesToolbarItems toolbarbutton.bookmark-item menupopup.toolbar-menupopup menuseparator::before {
    border-top: 1px solid var(--panel-separator-color) !important;
    content: "" !important;
    display: block !important;
    flex: 1 !important;
  }

  #PlacesToolbarItems toolbarbutton.bookmark-item menupopup.toolbar-menupopup menuseparator {
    appearance: unset !important;
    -moz-default-appearance: unset !important;
    border-top: unset !important;
    border-bottom: unset !important;
  }

  #bookmarks-menu-button menupopup#BMB_bookmarksPopup menuseparator::before {
    border-top: 1px solid var(--panel-separator-color) !important;
    content: "" !important;
    display: block !important;
    flex: 1 !important;
  }

  #bookmarks-menu-button menupopup#BMB_bookmarksPopup menuseparator {
    appearance: unset !important;
    -moz-default-appearance: unset !important;
    border-top: unset !important;
    border-bottom: unset !important;
  }
}

```

The last two rules fix the Bookmark Menu if you've placed it on your toolbar.

![image](https://github.com/Eclipse-Community/r3dfox/assets/625325/783ed107-3799-42d4-b129-ad1323b1483d)![image](https://github.com/Eclipse-Community/r3dfox/assets/625325/c99bb745-5471-42ad-8491-2b1425f7faad)

**Comment by K4sum1** [comment]
I might add that to menu.css then since it seems conditional and fixes the issue.

**Comment by K4sum1** [comment]
Added for 126.0.1, releasing soon.

**Comment by K4sum1** [comment]
If anyone is still interested, I realized there already is a method to get the Windows 10 UI on r3dfox on any Windows version.

Go to about:config and set widget.ev-native-controls-patch.override-win-version to 10

![r3dfox win 10 ui on 7](https://github.com/user-attachments/assets/db706ccf-73fe-4b1b-a227-4313b9e317d0)

**Comment by i486** [comment]
**Bug 1**: Since [Release v139.0.4rc for Vista+](https://github.com/Eclipse-Community/r3dfox/releases/tag/v139.0.4rc), I can no longer enable the non-native style. Tried unchecking "use native styles" and "Use r3dfox style", and checking "Windows Version Theme Override: Windows10", etc.

| v139.0.4rc | v139.0.3 |
|----------|---------|
| <img width="170" height="142" alt="Image" src="https://github.com/user-attachments/assets/37c60373-301e-48ac-bd2e-afe3108d8abd" />" /> | <img width="174" height="163" alt="Image" src="https://github.com/user-attachments/assets/31b4c4bb-9c0f-4ca0-9309-25d9845442f5" />" /> |

**Bug 2**: Since [Release v139.0.4-2 for Vista+](https://github.com/Eclipse-Community/r3dfox/releases/tag/v139.0.4-2), the appearance of the Bookmark and History sidebar has degraded.

| Expected | Current |
|----------|---------|
| <img width="247" height="737" src="https://github.com/user-attachments/assets/d9a7e5ae-1541-4383-9ac1-1a1185c973e4" /> | <img width="285" height="741" src="https://github.com/user-attachments/assets/e6e8b69d-00c2-4769-9274-e7895960f41f" /> |

Both bugs were tested with [Forest theme](https://addons.mozilla.org/en-US/firefox/addon/forest_/), but they also occur with other themes.

**Comment by K4sum1** [comment]
> **Bug 1**: Since [Release v139.0.4rc for Vista+](https://github.com/Eclipse-Community/r3dfox/releases/tag/v139.0.4rc), I can no longer enable the non-native style. Tried unchecking "use native styles" and "Use r3dfox style", and checking "Windows Version Theme Override: Windows10", etc.

I think I changed that theming to rely on browser.display.windows.non_native_menus instead of native controls. I think setting that to 1 or 0 will give you the Windows 10 menus everywhere.

> Bug 2: Since [Release v139.0.4-2 for Vista+](https://github.com/Eclipse-Community/r3dfox/releases/tag/v139.0.4-2), the appearance of the Bookmark and History sidebar has degraded.

I have an idea of what might have caused that. Will see if I can fix in 140.
